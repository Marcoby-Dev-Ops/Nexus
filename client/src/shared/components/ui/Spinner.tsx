import React from 'react';
import PropTypes from 'prop-types';

/**
 * @interface SpinnerProps
 * @description Props for the Spinner component.
 */
export interface SpinnerProps {
  size?: number;
  className?: string;
}

/**
 * @name Spinner
 * @description A spinner/loader component for loading states.
 * @param {SpinnerProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Spinner component.
 */
export const Spinner: React.FC<SpinnerProps> = React.memo(({ size = 24, className = '' }) => (
  <svg
    className={`animate-spin text-primary ${className}`}
    width={size}
    height={size}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http: //www.w3.org/2000/svg"
    role="status"
    aria-label="Loading"
  >
    <circle className="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="4" />
    <path className="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8v8z" />
  </svg>
));

Spinner.displayName = 'Spinner';

Spinner.propTypes = {
  size: PropTypes.number,
  className: PropTypes.string,
};

export default Spinner; 
