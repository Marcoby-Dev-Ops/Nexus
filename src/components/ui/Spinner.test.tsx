/**
 * @file Spinner.test.tsx
 * @description Unit and snapshot tests for the Spinner component.
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Spinner from './Spinner';

describe('Spinner', () => {
  it('renders without crashing', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 