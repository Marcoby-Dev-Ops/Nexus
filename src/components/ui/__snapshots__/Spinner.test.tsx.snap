// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-label="Loading"
    class="animate-spin text-primary "
    fill="none"
    height="24"
    role="status"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="opacity-25"
      cx="12"
      cy="12"
      r="10"
      stroke="currentColor"
      stroke-width="4"
    />
    <path
      class="opacity-75"
      d="M4 12a8 8 0 018-8v8z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;
